dlt/common/time.py

```diff
@@ -154,6 +155,29 @@ def ensure_pendulum_time(value: Union[str, datetime.time]) -> pendulum.Time:
     raise TypeError(f"Cannot coerce {value} to a pendulum.Time object.")
 
 
+def detect_datetime_format(value: str) -> Optional[str]:
+    format_patterns = {
```

Comment:
this is cool, but if you go beyond just parsing ISO date time, please unit-test a few cases

Reply:
Added unit-test, and extended ISO support.

Reply:
amazing :)